crates/sui-aws-orchestrator/src/protocol/narwhal.rs

```diff
@@ -109,14 +109,27 @@ impl ProtocolCommands<NarwhalBenchmarkType> for NarwhalProtocol {
     fn node_command<I>(
         &self,
         instances: I,
-        _parameters: &BenchmarkParameters<NarwhalBenchmarkType>,
+        parameters: &BenchmarkParameters<NarwhalBenchmarkType>,
     ) -> Vec<(Instance, String)>
     where
         I: IntoIterator<Item = Instance>,
     {
         let working_dir = self.working_dir.clone();
+        let hosts: Vec<_> = instances.into_iter().collect();
+        // 2 ports used per authority so add 2 * num authorities to base port
+        let mut worker_base_port = BASE_PORT + (2 * hosts.len());
```

Comment:
Why ports have to be different for each authority? Is it for supporting running on a single host?